src/api/app/controllers/webui/packages/binaries_controller.rb

```diff
@@ -60,6 +60,41 @@ def show
         end
       end
 
+      def dependency
+        @fileinfo = Backend::Api::BuildResults::Binaries.fileinfo_ext(@dependant_project_name, '_repository', @dependant_repository_name,
+                                                                      @architecture, params[:dependant_name])
+        return if @fileinfo # avoid displaying an error for non-existing packages
+
+        redirect_back(fallback_location: project_package_repository_binary_url(project_name: @project, package_name: @package,
+                                                                               repository: @repository, arch: @architecture, filename: @filename))
+      end
+
+      private
+
+      def set_dependant_project
+        @dependant_project_name = params[:dependant_project]
+        @dependant_project = Project.find_by_name(@dependant_project_name) || Project.find_remote_project(@dependant_project_name).try(:first)
+        return @dependant_project if @dependant_project
+
+        flash[:error] = "Project '#{elide(@dependant_project_name)}' is invalid."
+        redirect_back(fallback_location: root_path)
+      end
+
+      def set_dependant_repository
+        @dependant_repository_name = params[:dependant_repository]
+        # FIXME: It can't check repositories of remote projects
+        @dependant_repository = @dependant_project.repositories.find_by(name: @dependant_repository_name) if @dependant_project.remoteurl.blank?
+        return @dependant_repository if @dependant_repository
+
+        flash[:error] = "Repository '#{@dependant_repository_name}' is invalid."
+        redirect_back(fallback_location: project_show_path(project: @project.name))
+      end
+
+      def set_filename
+        # Ensure it really is just a file name, no '/..', etc.
+        @filename = File.basename(params[:binary_filename] || params[:filename])
```

Comment:
```suggestion
        @filename = File.basename(params[:filename])
```
I would not rename some ocurrences of the `filename` parameter to `binary_filename` in this commit. Or even include the renaming in this pull request.

Is it really needed this renaming to split out the `dependency` method?

Reply:
I guess we should add a parameter constraint , for `binary_filename` like this one: https://github.com/openSUSE/open-build-service/blob/296c43015231da84fdfd24871f9b61b957b72505/src/api/app/lib/routes_helper/routes_constraints.rb#L6